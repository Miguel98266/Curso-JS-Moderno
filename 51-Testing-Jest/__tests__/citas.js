import Citas from '../js/classes/Citas';

describe('Probar la clase Citas', () => {

    const citas = new Citas();
    const id=Date.now();
    test('Agregar una nueva cita', () => {
        const citaObj = {
            id,
            mascota: 'Hook',
            propietario: 'Miguel',
            telefono: '123213',
            fecha: '10-12-2021',
            hora: '10:30',
            sintomas: 'Solo duerme'
        }
       

        citas.agregarCita(citaObj);

        // Prueba
        expect(citas).toMatchSnapshot();
    });
    
    test('Actualizar cita',()=>{
        const citaActualizada = {
            id,
            mascota: 'Nuevo Nombre',
            propietario: 'Miguel',
            telefono: '123213',
            fecha: '10-12-2021',
            hora: '10:30',
            sintomas: 'Solo duerme'
        };
        citas.editarCita(citaActualizada);

        expect(citas).toMatchSnapshot();
    });

    test('Eliminar cita',()=>{
        citas.eliminarCita(id);

        expect(citas).toMatchSnapshot();
    });
});