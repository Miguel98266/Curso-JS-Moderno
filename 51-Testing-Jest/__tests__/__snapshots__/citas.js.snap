// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probar la clase Citas Actualizar cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-12-2021",
      "hora": "10:30",
      "id": 1633247064582,
      "mascota": "Nuevo Nombre",
      "propietario": "Miguel",
      "sintomas": "Solo duerme",
      "telefono": "123213",
    },
  ],
}
`;

exports[`Probar la clase Citas Agregar una nueva cita 1`] = `
Citas {
  "citas": Array [
    Object {
      "fecha": "10-12-2021",
      "hora": "10:30",
      "id": 1633247064582,
      "mascota": "Hook",
      "propietario": "Miguel",
      "sintomas": "Solo duerme",
      "telefono": "123213",
    },
  ],
}
`;

exports[`Probar la clase Citas Eliminar cita 1`] = `
Citas {
  "citas": Array [],
}
`;
